sdv/cag/base.py

```diff
@@ -45,6 +45,9 @@ def validate(self, data=None, metadata=None):
         if data is not None:
             if self._single_table:
                 data = {self._table_name: data}
+            elif isinstance(data, pd.DataFrame):
```

Comment:
This is necessary when running `validate` before fitting, since `self._single_table` wouldn't have been set yet.